Mathlib/AlgebraicTopology/SimplicialSet/HomotopyCat.lean

```diff
@@ -0,0 +1,469 @@
+  obj V := Cat.of (SSet.hoCat V)
+  map {S T} F := SSet.hoFunctorMap F
+  map_id S := by
+    apply Quotient.lift_unique'
+    simp [hoFunctorMap, Quotient.lift_spec]
+    exact Eq.trans (Functor.id_comp ..) (Functor.comp_id _).symm
+  map_comp {S T U} F G := by
+    apply Quotient.lift_unique'
+    simp [hoFunctorMap]
+    rw [Quotient.lift_spec, Cat.comp_eq_comp, Cat.comp_eq_comp, ← Functor.assoc, Functor.assoc,
+      Quotient.lift_spec, Functor.assoc, Quotient.lift_spec]
+end
+end
+
+section
+
+local macro:1000 (priority := high) X:term " _[" n:term "]₂" : term =>
+    `(($X : SSet.Truncated 2).obj (Opposite.op ⟨SimplexCategory.mk $n, by decide⟩))
+
+-- FIXME why doesn't this work?
+-- local notation3:1000 (priority := high) (prettyPrint := false) " _[" n "]₂" =>
+--     (X : SSet.Truncated 2).obj (Opposite.op ⟨SimplexCategory.mk n, by decide⟩)
+
+set_option quotPrecheck false
+local macro:max (priority := high) "[" n:term "]₂" : term =>
+  `((⟨SimplexCategory.mk $n, by decide⟩ : SimplexCategory.Truncated 2))
+
+/-- A 2-truncated simplicial set `S` has an underlying refl quiver with `S _[0]₂` as its underlying
+type.-/
+def SSet.OneTruncation₂ (S : SSet.Truncated 2) := S _[0]₂
+
+/-- Abbreviations for face maps in the 2-truncated simplex category.-/
+abbrev SSet.δ₂ {n} (i : Fin (n + 2)) (hn := by decide) (hn' := by decide) :
+    (⟨[n], hn⟩ : SimplexCategory.Truncated 2) ⟶ ⟨[n + 1], hn'⟩ := SimplexCategory.δ i
+
+/-- Abbreviations for degeneracy maps in the 2-truncated simplex category.-/
+abbrev SSet.σ₂ {n} (i : Fin (n + 1)) (hn := by decide) (hn' := by decide) :
+    (⟨[n+1], hn⟩ : SimplexCategory.Truncated 2) ⟶ ⟨[n], hn'⟩ := SimplexCategory.σ i
+
+/-- The source vertex of `f : S _[1]₂` for use in defining the underlying refl quiver.-/
+def SSet.OneTruncation₂.src {S : SSet.Truncated 2} (f : S _[1]₂) : OneTruncation₂ S :=
+  S.map (δ₂ (n := 0) 1).op f
+
+/-- The target vertex of `f : S _[1]₂` for use in defining the underlying refl quiver.-/
+def SSet.OneTruncation₂.tgt {S : SSet.Truncated 2} (f : S _[1]₂) : OneTruncation₂ S :=
+  S.map (δ₂ (n := 0) 0).op f
+
+/-- The hom-types of the refl quiver underlying a simplicial set `S` are subtypes of `S _[1]₂`.-/
+def SSet.OneTruncation₂.Hom {S : SSet.Truncated 2} (X Y : OneTruncation₂ S) :=
+  {p : S _[1]₂ // src p = X ∧ tgt p = Y}
+
+/-- A 2-truncated simplicial set `S` has an underlying refl quiver `SSet.OneTruncation₂ S`.-/
+instance (S : SSet.Truncated 2) : ReflQuiver (SSet.OneTruncation₂ S) where
+  Hom X Y := SSet.OneTruncation₂.Hom X Y
+  id X := by
+    refine ⟨S.map (SSet.σ₂ (n := 0) 0).op X, ?_, ?_⟩ <;>
+    · change (S.map _ ≫ S.map _) X = X
+      rw [← map_comp]
+      rw [(_ : _ ≫ _ = 𝟙 _)]; simp
+      show ({..} : Opposite _) = _; congr; dsimp [SimplexCategory.Truncated]; ext ⟨i, _⟩
+      let 0 := i
+      rfl
+
+/-- The functor that carries a 2-truncated simplicial set to its underlying refl quiver.-/
+def SSet.oneTruncation₂ : SSet.Truncated.{u} 2 ⥤ ReflQuiv.{u,u} where
```

Comment:
It would be really nice to avoid duplication of code here. The only solution I see is to define the reflexive quiver and relations only in the case of a `S : SSet.truncated 2` and then, when we want to apply the construction to fully fledged simplicial sets, we just apply the truncation first. This may require developing the basic API of simplicial relations suggested above for the truncated simplicial objects/simplicial sets as well.

Reply:
We think it would be better not to deal with this in this PR, but rather wait until we have some usage code before attempting such a refactor, because `OneTruncation` is not an opaque definition and we would prefer for users not to have to deal with the 2-truncation layer when they don't explicitly need it. Refactoring now could be making things harder elsewhere in code that does not exist yet.